warrior/Actions/CommonActions/common_actions.py

```diff
@@ -198,30 +169,44 @@ def verify_key_already_exists_and_update(orig_dict, new_dict):
                 filepath = getAbsPath(filepath, os.path.dirname(testcasefile_path))
                 with open(filepath, "r") as json_handle:
                     json_doc = json.load(json_handle)
-                    if jsonkey in json_doc:
-                        dict_to_update = {}
-                        repo_dict = json_doc[jsonkey]
-                        for var_key, var_value in repo_dict.items():
-                            if isinstance(var_value, dict):
-                                if var_value.get('type') == 'int':
-                                    value = int(var_value['value'])
-                                elif var_value.get('type') == 'float':
-                                    value = float(var_value['value'])
-                                else:
-                                    value = str(var_value['value'])
+                #if bool_store_all is set to True, all content of given json file will be
+                #stored in data repository
+                if isinstance(bool_store_all, bool) and bool_store_all is True:
+                    print_info("bool_store_all is set to True, all content of given"
+                               " json file will be stored in data repository")
+                    update_datarepository(json_doc)
+                    print_info("{0} dictionary stored in Warrior data_repository".
+                               format(json_doc))
+                    status = True
+                elif not isinstance(bool_store_all, bool):
+                    print_error("invalid value : {0} given for bool_store_all,"
+                                "valid value: boolean True or False".format(bool_store_all))
+                    status = False
+                elif jsonkey in json_doc:
```

Comment:
The above statement needs to be:
``` elif json_doc and jsonkey in json_doc``` and json_doc needs to be initialized before the ```with``` statement on line 170 as ```json_doc = {}```

Reply:
this change is not required as with() block will raise an exception if file is not present or invalid json is given and control never reaches below 'if elif blocks' unless json_doc is defined.

you may check the last testcase (negative scenarios) which tests these functionality in given travis log.
https://api.travis-ci.org/v3/job/479250372/log.txt